lib/libyear_bundler/models/ruby.rb

```diff
@@ -0,0 +1,133 @@
+require 'bundler/lockfile_parser'
+require 'bundler/ruby_version'
+require 'date'
+require 'net/http'
+require 'open3'
+require 'yaml'
+
+require 'libyear_bundler/calculators/libyear'
+require 'libyear_bundler/calculators/version_number_delta'
+require 'libyear_bundler/calculators/version_sequence_delta'
+
+module LibyearBundler
+  module Models
+    class Ruby
+      def initialize(lockfile)
+        @lockfile = lockfile
+      end
+
+      def installed_version
+        @_installed_version ||= begin
+          version_from_bundler ||
+            version_from_ruby_version_file ||
+            version_from_ruby
+        end
+      end
+
+      def installed_version_release_date
+        release_date(installed_version.to_s)
+      end
+
+      def libyears
+        ::LibyearBundler::Calculators::Libyear.calculate(
+          release_date(installed_version.to_s),
+          release_date(newest_version.to_s)
+        )
+      end
+
+      def name
+        'ruby'
+      end
+
+      def newest_version
+        ::Gem::Version.new(all_versions.first['version'])
+      end
+
+      def newest_version_release_date
+        release_date(newest_version.to_s)
+      end
+
+      def outdated?
+        installed_version < newest_version
+      end
+
+      def version_number_delta
+        ::LibyearBundler::Calculators::VersionNumberDelta.calculate(
+          installed_version,
+          newest_version
+        )
+      end
+
+      def version_sequence_delta
+        ::LibyearBundler::Calculators::VersionSequenceDelta.calculate(
+          installed_version_sequence_index,
+          newest_version_sequence_index
+        )
+      end
+
+      private
+
+      # We'll only consider non-prerelease versions when determining the
+      # newest version
+      #
+      # The following URL is the only official, easily-parseable document with
+      # Ruby version information that I'm aware of, but is not supported as such
+      # (https://github.com/ruby/www.ruby-lang.org/pull/1637#issuecomment-344934173).
+      # It's been recommend that ruby-lang.org provide a supported document:
+      # https://github.com/ruby/www.ruby-lang.org/pull/1637#issuecomment-344934173
+      # TODO: Use supported document with version information if it becomes
+      # available.
+      def all_versions
+        @_all_versions ||= begin
+          uri = ::URI.parse("https://raw.githubusercontent.com/ruby/www.ruby-lang.org/master/_data/releases.yml")
```

Comment:
Nice job finding this data!

This is good for now. In the future we may want to commit this file so we have something to fall back on if the online version ever goes away.